### DISCUSSION TITLE
Outgoing Ports Required for Skate

### DISCUSSION THREAD

**Discussion by ProfessorManhattan** [charmbracelet/skate#75] [Q&A] [answered]
Hey, I have outbound firewall rules and would like to add one for Skate. What UDP ports and TCP ports are required for Skate?

**Comment by ProfessorManhattan** [comment]
nevermind, `2408` appears to be first one that is used (along with a few others that shouldn't matter if 2408 is opened). Upon opening 2408, I am seeing 255.255.255.255:67 --- what is Skate multicasting for?

**Comment by bashbunni** [comment, marked as answer]
Hey @ProfessorManhattan Skate is essentially a layer over the Charm KV data store, so all of the config for Charm affects Skate as well. On that note, [here](https://github.com/charmbracelet/charm#client-settings) is the list of client-side settings you can change. 

Our docs for self-hosting Charm should help to clarify the multicasting and port setup for that's happening under the hood  https://github.com/charmbracelet/charm/blob/main/docs/self-hosting.md#self-hosting-with-tls

Let me know if that answers your questions :) It's my understanding that the multicasting is to support both HTTP and SSH requests as mentioned in the self-hosting docs.
